eZ/Publish/API/Repository/Tests/UserServiceTest.php

```diff
@@ -1576,6 +1578,83 @@ public function testUpdateUser()
     /**
      * Test for the updateUser() method.
      *
+     * @return \eZ\Publish\API\Repository\Values\User\User
+     *
+     * @see \eZ\Publish\API\Repository\UserService::updateUser()
+     * @depends eZ\Publish\API\Repository\Tests\UserServiceTest::testCreateUser
+     * @depends eZ\Publish\API\Repository\Tests\UserServiceTest::testNewUserUpdateStruct
+     * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testUpdateContent
+     * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testUpdateContentMetadata
+     */
+    public function testUpdateUserNoPassword()
+    {
+        $repository = $this->getRepository();
+        $signalSlotUserService = $repository->getUserService();
+
+        $signalSlotUserServiceReflection = new ReflectionClass($signalSlotUserService);
+        $userServiceProperty = $signalSlotUserServiceReflection->getProperty('service');
+        $userServiceProperty->setAccessible(true);
+        $userService = $userServiceProperty->getValue($signalSlotUserService);
+
+        $userServiceReflection = new ReflectionClass($userService);
+        $settingsProperty = $userServiceReflection->getProperty('settings');
+        $settingsProperty->setAccessible(true);
+        $settingsProperty->setValue(
+            $userService,
+            [
+                'hashType' => User::PASSWORD_HASH_MD5_USER,
+            ] + $settingsProperty->getValue($userService)
+        );
+
+        /* BEGIN: Use Case */
+        $user = $this->createUserVersion1();
+
+        $settingsProperty->setValue(
+            $userService,
+            [
+                'hashType' => User::PASSWORD_HASH_PHP_DEFAULT,
+            ] + $settingsProperty->getValue($userService)
+        );
```

Comment:
Note to self: Uh, we should consider exposing a possibility to update settings via API in the future.

Reply:
That would be useful here, yes :)